cdap-app-fabric/src/main/java/org/apache/hadoop/mapred/LocalDistributedCacheManagerWithFix.java

```diff
@@ -107,7 +107,7 @@ public void setup(JobConf conf) throws IOException {
     MRApps.setupDistributedCache(conf, localResources);
     // Generating unique numbers for FSDownload.
     AtomicLong uniqueNumberGenerator =
-      new AtomicLong(System.currentTimeMillis());
+      new AtomicLong(System.nanoTime());
```

Comment:
The only problem is nanoTime has no relationship with the actual time. Can you apply the suggestion as mentioned in 

https://issues.apache.org/jira/browse/HADOOP-10924?focusedCommentId=14323087&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14323087

?